Sources/Private/Experimental/ExperimentalAnimationLayer.swift

```diff
@@ -75,6 +80,13 @@ final class ExperimentalAnimationLayer: CALayer {
     for animationLayer in animationLayers {
       animationLayer.setupAnimations(context: context)
     }
+
+    // We also set up a placeholder animation to track the current animation progress
+    let animationProgressTracker = CABasicAnimation(keyPath: #keyPath(animationProgress))
+    animationProgressTracker.configureTiming(with: context)
+    animationProgressTracker.fromValue = 0
+    animationProgressTracker.toValue = 1
+    add(animationProgressTracker, forKey: key)
```

Comment:
Is this going to add some overhead to these animations as this will ping the main thread at the device frame rate? Wonder if we could do something wild like have a size 0 layer animating its opacity and then getting its `presentationLayer().opacity`?

Reply:
Good question, since this doesn't have `needsDisplay: true` I assume it doesn't actually run on the main thread every frame. Let me double-check this though.

Reply:
When I add a `didSet` tp `animationProgress` and set a breakpoint, it doesn't get called while the animation is playing. CPU usage is also stays at 0% while playing the animation.

Reply:
Nice! Great to hear